### PR TITLE
Ingress controller: add support for ingress class annotation

### LINKED ISSUE #215: Ingress controller: support ingress class annotation
Ingress resources can have a `kubernetes.io/ingress.class` annotation (see [documentation](https://github.com/kubernetes/ingress/blob/master/examples/PREREQUISITES.md#ingress-class)), which, if present, indicates the name of the ingress controller which should own/process the ingress resource.

This is mainly relevant for clusters in which both a user-provided and a cloud-provider ingress controllers are run together.

### PR DESCRIPTION
**Pull request by zcahana**
Fixes #215.

### CODE REVIEW

File: platform/kube/controller_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	"k8s.io/client-go/pkg/apis/extensions/v1beta1"
+	"k8s.io/client-go/pkg/util/intstr"
+)
+
+func TestIngressIgnored(t *testing.T) {
+	cases := []struct {
+		ingressClass string
+		shouldIgnore bool
+	}{
+		{ingressClass: "nginx", shouldIgnore: true},
```
Comment (rshriram) [inline platform/kube/controller_test.go]: How do we ensure that when we chain ingress controllers, other controllers respect such annotations? I guess it's out of our hand.

Comment (zcahana) [inline platform/kube/controller_test.go]: We cannot ensure that others are well-behaving, but we can ensure we are.
For what it's worth, GKE and Nginx (both community and Nginx Inc) ingress controllers respect it.

**Comment by rshriram** [comment]
Please document before you merge!

**Comment by rshriram** [comment]
Or after. May be stick an example ingress doc in istio/istio. We can organize later.

File: platform/kube/controller_test.go (hunk 1 above)
Comment (kyessenov) [inline platform/kube/controller_test.go]: Yes, this is the right way to separate controllers from acting on the same ingress resource.

**Comment by zcahana** [comment]
Opened #241 for doc'ing the ingress controller stuff.

**Pull request merged**